src/test/ui/closures/2229_closure_analysis/diagnostics/closure-origin-tuple-diagnostics.rs

```diff
@@ -0,0 +1,15 @@
+#![feature(capture_disjoint_fields)]
+//~^ WARNING: the feature `capture_disjoint_fields` is incomplete
+//~| `#[warn(incomplete_features)]` on by default
+//~| see issue #53488 <https://github.com/rust-lang/rust/issues/53488>
+struct S(String);
+
+fn expect_fn<F: Fn()>(_f: F) {}
+
+fn main() {
+    let s = S(format!("s"));
+    let c = || { //~ ERROR expected a closure
```

Comment:
can u turn the error into expected closure that implements `Fn`

Reply:
Also can you modify the test like this and use `s.1`: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=5c03fa69ecf41fc95e30bf12c937a848

Reply:
ohh, I see what you want now